packages/map-template/src/atoms/currentKioskLocationState.js

```diff
@@ -0,0 +1,8 @@
+import { atom } from 'recoil';
+
+const currentKioskLocationState = atom({
```

Comment:
Does this contain the whole Location or just the ID? Can it be merged with the `kioskOriginLocationIdState`?

Reply:
The `currentKioskLocationState` contains the whole location, which we need for getting directions. So we need that.

Reply:
Can we then use this one instead of `kioskOriginLocationIdState`?

Reply:
Hmm, I'm not sure we can, because we need to first listen to the changes in the kioskOriginLocationId prop, and then based on that id, we get the location and then set the setCurrentKioskLocation on line 262.

Reply:
Yes. We already listen for changes in the `kioskOriginLocationId` prop. Reacting to that, we can read the location and store it in an atom. And use that atom wherever we now use the kioskLocationId atom. That way we will only have one atom to store kiosk origin info instead of two. Wouldn't that be cleaner?

Reply:
Removed the `travelMode` and `accessibilityOn` dependencies from the useEffect, as they are passed through props.

Reply:
Does that fix it? 🙏